test/inline_asm_basic.cl

```diff
@@ -0,0 +1,19 @@
+// RUN: %clang_cc1 -triple spir64-unknown-unknown -x cl -cl-std=CL2.0 -O0 -fdeclare-opencl-builtins -finclude-default-header -emit-llvm-bc %s -o %t.bc
+// RUN: llvm-spirv %t.bc -spirv-ext=+SPV_INTEL_inline_assembly -spirv-text -o %t.spt
+// RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+// RUN: llvm-spirv -spirv-ext=+SPV_INTEL_inline_assembly %t.bc -o %t.spv
+// RUN: spirv-val %t.spv
+// RUN: llvm-spirv -spirv-ext=+SPV_INTEL_inline_assembly -r %t.spv -o %t.bc
```

Comment:
```suggestion
// RUN: llvm-spirv -spirv-ext=+SPV_INTEL_inline_assembly %t.bc -o %t.spv
// RUN: llvm-spirv %t.spv -to-text -o %t.spt
// RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
// RUN: spirv-val %t.spv
// RUN: llvm-spirv -r %t.spv -o %t.bc
```

I suggest to generate text representation from binary representation: conversion from binary to text doesn't require enabling any extensions.

Also, while converting back to LLVM IR, all extensions are enabled by default